### DISCUSSION TITLE
Trying to run OpenDDS federate DCPSInfoRepo

### DISCUSSION THREAD

**Discussion by jlseminara** [OpenDDS/OpenDDS#3721] [Q&A] [answered]
I'm trying to run 4 federated DCPSInfoRepo but couldn't make it. The "OpenDDS Dev Guide" is not of much help

Does anyone have any clue?

**Comment by mitza-oci** [comment, marked as answer]
What, specifically, did you try (add complete command lines, configuration files, etc.)?  What was observed?  How did that differ from what you expected?

**Reply by jlseminara** [reply]
So sorry for being so vague in my description, I'm really newbie at OpenDDS. I don`t have a base of code yet. I'm just trying to run 4 federated DCPSInfoRepo via command line. (Don't even know if that is possible)

**Reply by jlseminara** [reply]
I also try to communicate published and subscriber via RTPS but the subscriber couldn't find the publisher in different machines. It works perfectly on the same machine.  This is the config file I'm using

[common]
DCPSGlobalTransportConfig=$file
DCPSDefaultDiscovery=DEFAULT_RTPS

[transport/the_rtps_transport]
transport_type=rtps_udp

**Reply by simpsont-oci** [reply]
For RTPS, the machines need to be able to send multicast traffic to one another. For a lot of machines, this can mean disabling the firewall if you have one enabled. For the sake of debugging, it would be helpful if you included the name of the example or test you were running (there are an awful lot of publishers and subscribers in our tests) as well as the info Adam requested (full command lines, config files, output & logs). And for RTPS in particular, it might be helpful to check to make sure you can see the RTPS/UDP traffic for both machines using a tool like tcpdump (using the udp flag) or Wireshark (and then you could include the packet capture file as well).

**Reply by jlseminara** [reply]
Thank you so much for you reply Timothy.  I'm guessing the problem is the network configuration. I'm using two virtual machines on Azure I don't think UDP Multicast is enabled.

**Reply by simpsont-oci** [reply]
You can partially get around the multicast issue by disabling multicast in the discovery and transport configurations (see `SedpMulticast` and `use_multicast` flags, respectively) but you will also likely need to uniquely specify `SpdpSendAddrs` and `SpdpLocalAddress` for all machines involved (so the config file would no longer be common between machines).

**Reply by mitza-oci** [reply]
This is a different topic, it has nothing to do with "federate DCPSInfoRepo"

But please see https://github.com/oci-labs/cloud-multicast-repeater which we developed for handling deployment environments that lack multicast.

**Reply by jlseminara** [reply]
Ok, understood. Unfortunately I need full discovery without specifying all the participants. I'll see what I can do. 

Thank you so much again Timothy. You've been of much help

**Reply by jrw972** [reply]
The RtpsRelay is another option.  https://objectcomputing.com/resources/publications/mnb/2019/06/20/interoperable-internet-enabled-dds-applications
